src/azul/__init__.py

```diff
@@ -745,7 +745,10 @@ def domain_name(self) -> str:
 
     @property
     def private_api(self) -> bool:
-        return self._boolean(self.environ['AZUL_PRIVATE_API'])
+        is_private_api = self._boolean(self.environ['AZUL_PRIVATE_API'])
+        if self.is_anvil_enabled():
+            require(is_private_api)
+        return is_private_api
```

Comment:
```suggestion
        is_private_api = self._boolean(self.environ['AZUL_PRIVATE_API'])
        if not is_private_api:
            reject(self.is_anvil_enabled(), 'AnVIL deployments must be private')
        return is_private_api
```

… considering that `is_anvil_enabled` is fairly expensive.